### PR TITLE
Fixup demo app and export all types

### PR DESCRIPTION
**Pull request by matthieubosquet**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,14 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
...
 
+### Bugfix
+
+- Export all types via Rollup.
+- Fix demo application.
+
 ### New features
 
--
+- Drop Node 12 support.
```
Comment (ThisIsMissEm) [inline CHANGELOG.md:16]: This is also a breaking change, not a new feature.

Comment (matthieubosquet) [reply CHANGELOG.md:16]: Marked as such.

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -136,7 +136,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        node-version: [16.x, 14.x, 12.x]
+        node-version: ["18", "16", "14"]
```
[the remark this answers is not in the document]
Comment (matthieubosquet) [reply .github/workflows/cd.yml]: Removed

File: examples/grant-access/views/grant-form.ejs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            <label for="deny">Deny</label>
+            <input type="radio" name="manage" id="deny" value="deny" required />
+            <input id="requestVc" type="hidden" name="requestVc" value='<%= JSON.stringify(accessRequest) %>'>
+            <input id="redirectUrl" type="hidden" name="redirectUrl" value="<%- requestorRedirectUrl %>">
+        </p>
+        <p>
+            <input type="submit" value="Respond to Access Request">
+        </p>
+    </fieldset>
+</form>
```
Comment (ThisIsMissEm) [inline examples/grant-access/views/grant-form.ejs:17]: missing newline at EOF

File: src/index.ts [hunk 4]
```diff
@@ -19,18 +19,22 @@
...
 
-export type { AccessModes } from "./type/AccessModes";
 export type { AccessBaseOptions } from "./type/AccessBaseOptions";
+export type { AccessCredentialType } from "./type/AccessCredentialType";
+export type { AccessGrantContext } from "./type/AccessGrantContext";
+export type { AccessModes } from "./type/AccessModes";
+export type { ApproveAccessRequestOverrides } from "./manage/approveAccessRequest";
+export type { FetchOptions } from "./type/FetchOptions";
 export type { IssueAccessRequestParameters } from "./type/IssueAccessRequestParameters";
+export type { RedirectOptions } from "./type/RedirectOptions";
```
Comment (ThisIsMissEm) [inline src/index.ts:29]: Might be good to merge this type into `RedirectToAccessManagementUiOptions` rather than having it seperate?

**Review by ThisIsMissEm — APPROVED** [review]
I'm not confident on the changes made to the root tsconfig files, especially changing the target from es2015/2017 to es2020; I think we do still want to output for es2015 for browsers.

I'm going to approve, but it'd be better to make those changes in a more scheduled fashion when we've time to reason about what the implications are.

File: tsconfig.json [hunk 5]
```diff
@@ -1,23 +1,20 @@
 {
   "compilerOptions": {
     /* Basic Options */
-    "target": "es2017",
+    "target": "es2020",
```
Comment (matthieubosquet) [inline tsconfig.json:4]:
```suggestion
    "target": "es2017",
```

File: examples/grant-access/views/grant-form.ejs (hunk 3 above)
Comment (matthieubosquet) [reply examples/grant-access/views/grant-form.ejs:17]:
```suggestion
</form>

```

File: src/index.ts (hunk 4 above)
Comment (matthieubosquet) [reply src/index.ts:29]: Sounds good, but maybe let's address this in a separate PR.

Comment (ThisIsMissEm) [reply src/index.ts:29]: sure!

**Comment by matthieubosquet** [comment]
> I'm not confident on the changes made to the root tsconfig files, especially changing the target from es2015/2017 to es2020; I think we do still want to output for es2015 for browsers.
>
> I'm going to approve, but it'd be better to make those changes in a more scheduled fashion when we've time to reason about what the implications are.

I reverted to target `ES2017` as it was before. My general idea would be to target the maintenance Node LTS (14 is ES2020) and let parcel/browserify & co deal with older browser requirements.

**Comment by ThisIsMissEm** [comment]
> > I'm not confident on the changes made to the root tsconfig files, especially changing the target from es2015/2017 to es2020; I think we do still want to output for es2015 for browsers.
> > I'm going to approve, but it'd be better to make those changes in a more scheduled fashion when we've time to reason about what the implications are.
> 
> I reverted to target `ES2017` as it was before. My general idea would be to target the maintenance Node LTS (14 is ES2020) and let parcel/browserify & co deal with older browser requirements.

Yup, but that's something we can do in a planned manner, not as a small change in a larger PR, especially this close to a big release.

**Pull request merged**